optuna/samplers/_tpe/sampler.py

```diff
@@ -893,3 +891,30 @@ def _calculate_weights_below_for_multi_objective(
     weights_below_all[feasible_mask] = weights_below
     weights_below_all = weights_below_all[~np.isnan(cvals)]
     return weights_below_all
+
+
+def _clamp_inf_and_calc_reference_point(lvals: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
+    """Clamp inf values in lvals and calculate the reference point.
+
+    This function satisfies the following axioms:
+    1. Clamped lvals and the calculated reference point only contain finite values.
+    2. Clamping preserves the original order of lvals.
+    3. Clamped lvals are always smaller than or equal to the calculated reference point.
+    """
```

Comment:
Could you add explanation that the clipped values do not affect the TPE algorithm? In my understanding, single-objective TPE only consider the order of objective values. Is the situation the same for the multi-objective case?

Reply:
Does this clipping algorithm keeps the expected hypervolume improvement?
IMO, the order of the trials can be changed depending on the scale of each parameter.

For example, when we have three points such as (`-inf`, -900), (-100, -1000) and (-90, -1200) in a target rank and a reference point (0, 0), the clipped points will be (-110, -900), (-100, -1000) and (-90, -1200).

The changes of the hypervolumes will be as follows:
Before clipping: `inf` (`inf` * 900), 1000 (10 * 100), 18000 (90 * 200)
After clipping: 9000 (10 * 900), 1000 (10 * 100), 18000 (90 * 200)

So, the rank of first point and third point will be changed:
Before clipping: 0, 2, 1
After clipping: 1, 2, 0

Reply:
Yes, it can change the order of the hypervolume improvement. This is intended, because we don't want all trials containing at least one `inf` to all have the same hypervolume improvement.

Reply:
This should not be a breaking change, because having `inf` as hypervolumes would result in `nan`.

Reply:
> In my understanding, single-objective TPE only consider the order of objective values. Is the situation the same for the multi-objective case?
No, as for multi-objective case, the weights depend on the hypervolume improvements. Thus, the clipped values DO affect the TPE algorithm.